tests/codegen/call_usym_key.cpp

```diff
@@ -17,35 +17,33 @@ declare void @llvm.lifetime.start.p0i8(i64, i8* nocapture) #1
 define i64 @"kprobe:f"(i8* nocapture readnone) local_unnamed_addr section "s_kprobe:f_1" {
 entry:
   %"@x_val" = alloca i64, align 8
-  %"@x_key" = alloca [16 x i8], align 8
-  %1 = getelementptr inbounds [16 x i8], [16 x i8]* %"@x_key", i64 0, i64 0
+  %usym = alloca [16 x i8], align 8
+  %1 = getelementptr inbounds [16 x i8], [16 x i8]* %usym, i64 0, i64 0
```

Comment:
Also here in the optimised usym case

Reply:
It seems LLVM changes bitcast to getelementptr after optimization (bitcast instruction is emitted along with lifetime.start/end.) So I think these getelemenptr instructions are OK.

```
Before optimization
------------------
define i64 @"kprobe:f"(i8*) section "s_kprobe:f_1" {
entry:
  %"@x_val" = alloca i64
  %lookup_elem_val = alloca i64
  %usym = alloca [16 x i8]
  %1 = bitcast [16 x i8]* %usym to i8*
  call void @llvm.lifetime.start.p0i8(i64 -1, i8* %1)
...
  %11 = bitcast [16 x i8]* %usym to i8*
  call void @llvm.lifetime.end.p0i8(i64 -1, i8* %11)
...
```
```
After optimization
------------------
...
define i64 @"kprobe:f"(i8* nocapture readnone) local_unnamed_addr section "s_kprobe:f_1" {
entry:
  %"@x_val" = alloca i64, align 8
  %usym = alloca [16 x i8], align 8
  %1 = getelementptr inbounds [16 x i8], [16 x i8]* %usym, i64 0, i64 0
  call void @llvm.lifetime.start.p0i8(i64 -1, i8* nonnull %1)
....
  call void @llvm.lifetime.end.p0i8(i64 -1, i8* nonnull %1)
...
```